### PR TITLE
[PWGCF] added support for IR and current run duration

### PR DESCRIPTION
**Pull request by abilandz**

### CODE REVIEW

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 1]
```diff
@@ -9697,6 +10047,68 @@ void DetermineOccupancy(T const& collision)
...
+                                                   // cout << bc.timestamp() << endl;  // works as expected
+                                                   // TBI 20241126 I have the same code snippet at 2 other locations, I shall unify at some point
+
+    // Get start timestamp and end timemstamp for this run in miliseconds, and convert both of them in seconds:
+    // o see O2/CCDB/src/BasicCCDBManager.cxx, O2/CCDB/include/CCDB/BasicCCDBManager.h
+    // o example usage in O2Physics/PWGLF/TableProducer/Common/zdcSP.cxx
+    auto runDuration = ccdb->getRunDuration(bc.runNumber()); // this is total run duration, not the current one (see below)
+    int64_t startOfRun = std::floor(runDuration.first * 0.001);
+    int64_t endOfRun = std::ceil(runDuration.second * 0.001);
+    int64_t runDurationInSec = endOfRun - startOfRun; // yes, this is now in seconds
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:10076]: Consider doing this at run level and not at collision level. They will not change, right?

**Review by victor-gonzalez — APPROVED** [review]
Have a look at my comment in case yo want to consider it for the next iteration

**Comment by victor-gonzalez** [comment]
@abilandz The linker is complaining about one of your functions in your `MuPa-MemberFunctions.h` not being resolved
Have you produce O2Physics locally with your changes?

**Comment by abilandz** [comment]
> @abilandz The linker is complaining about one of your functions in your `MuPa-MemberFunctions.h` not being resolved Have you produced O2Physics locally with your changes?

I wrote you earlier today email about that, here it is:

I just made a new pull request [#8710](https://github.com/AliceO2Group/O2Physics/pull/8710) , but I anticipate the compilation most likely will fail.  

Namely, to use Common/CCDB/ctpRateFetcher , I have to include both .h and .cxx files in the preamble of my main executable [file](https://github.com/AliceO2Group/O2Physics/pull/8710/commits/5365958eb5a220c3cfd8fcc1a4f382bc2141f82f#diff-e92f890001c4cee4fb82d17fcef239d37e5717cd4c06ebeca5107d44dea2f8da) :

#include "Common/CCDB/ctpRateFetcher.h"
#include "Common/CCDB/ctpRateFetcher.cxx" // TBI 20241126 if I include instead only .h file, linker fails, check this further

If I include only .h, the linker fails. 

However, if I include also .cxx, I can compile locally and run, but then MegaLinter fails when I made PR, as ML doesn't allow include's of .cxx files, apparently.

Can you please suggest how to proceed here?

Thanks!

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h (hunk 1 above)
Comment (abilandz) [reply PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:10076]: Yes, well-spotted, thanks!

**Comment by victor-gonzalez** [comment]
Ok! Sorry about that!
I was away the whole day and started with the pending PRs. Not read email yet!

You should never include the source file
You need to incorporate the proper library to the build of your executable/s

Go to the corresponding `CMakeLists.txt` of your considered executable and add the `O2Physics::AnalysisCCDB` library to the `PUBLIC_LINK_LIBRARIES` libraries list of your considered executable

**Comment by abilandz** [comment]
> Ok! Sorry about that! I was away the whole day and started with the pending PRs. Not read email yet!
> 
> You should never include the source file You need to incorporate the proper library to the build of your executable/s
> 
> Go to the corresponding `CMakeLists.txt` of your considered executable and add the `O2Physics::AnalysisCCDB` library to the `PUBLIC_LINK_LIBRARIES` libraries list of your considered executable

Thanks a lot Señor, noted! I will do and test it in the morning, and make new commit with this change!

**Comment by abilandz** [comment]
> > Ok! Sorry about that! I was away the whole day and started with the pending PRs. Not read email yet!
> > You should never include the source file You need to incorporate the proper library to the build of your executable/s
> > Go to the corresponding `CMakeLists.txt` of your considered executable and add the `O2Physics::AnalysisCCDB` library to the `PUBLIC_LINK_LIBRARIES` libraries list of your considered executable
> 
> Thanks a lot Señor, noted! I will do and test it in the morning, and make new commit with this change!

Done! Everything is OK now!

**Review by victor-gonzalez — APPROVED** [review]

**Pull request merged**